vim: Maintain block cursor for navigating/non-modifying operators

5brian opened the pull request:
The cursor shape now only changes to underline for operators that modify text (like: delete, change, yank) while maintaining block shape for navigation operators (like: find, till).

This better matches Vim/Nvim's behavior where the cursor only changes shape when an operator that will modify text is pending.

Release Notes:

- vim: Improved cursor shape behavior to better match Vim

5brian committed:
vim cursor

maxbrunsfeld commented:
From my testing, Neovim uses the underline cursor for other operators as well, like indent, dedent, auto-indent. Should we invert this logic so that we explicitly keep the block cursor for a select few navigational operators (`find`, `til`)?

5brian committed:
invert match logic

5brian committed:
rm literal operator

5brian committed:
add more navigation operators

5brian commented:
That makes more sense, thanks! Updated with the inverted logic, and added mark and macro operators to the navigational operators.

Might still be missing a few operators which should be block cursor that i am unaware of.

xzbdmw commented:
FWIW, these are all the operators that will change to underscore
```txt
The motion commands can be used after an operator command, to have the command
operate on the text that was moved over.  That is the text between the cursor
position before and after the motion.  Operators are generally used to delete
or change text.  The following operators are available:

	|c|	c	change
	|d|	d	delete
	|y|	y	yank into register (does not change the text)
	|~|	~	swap case (only if 'tildeop' is set)
	|g~|	g~	swap case
	|gu|	gu	make lowercase
	|gU|	gU	make uppercase
	|!|	!	filter through an external program
	|=|	=	filter through 'equalprg' or C-indenting if empty
	|gq|	gq	text formatting
	|gw|	gw	text formatting with no cursor movement
	|g?|	g?	ROT13 encoding
	|>|	>	shift right
	|<|	<	shift left
	|zf|	zf	define a fold
	|g@|	g@	call function set with the 'operatorfunc' option
```
